### PR TITLE
Add domains management

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes/no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: front/domaintype.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<<?php
```
Comment (AdrienClairembault) [inline front/domaintype.php]: Typo in a few files:

```suggestion
<?php
```
Also in `front/domainrelation.php` and `front/domainrecordtype.php`.

File: inc/domainrecord.class.php [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+      }
+      return true;
+   }
+
+   function rawSearchOptions() {
+      $tab = [];
+
+      $tab = array_merge($tab, parent::rawSearchOptions());
+
+      /*$tab[] = [
```
Comment (orthagh) [inline inc/domainrecord.class.php]: to clean ?

File: inc/domainrecord.class.php [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+      $this->addStandardTab('Item_Problem', $ong, $options);
+      $this->addStandardTab('Document_Item', $ong, $options);
+      $this->addStandardTab('Link', $ong, $options);
+      $this->addStandardTab('Notepad', $ong, $options);
+      $this->addStandardTab('Log', $ong, $options);
+
+      return $ong;
+   }
+
+   /*private function prepareInput($input) {
```
Comment (orthagh) [inline inc/domainrecord.class.php]: to clean ?

File: install/empty_data.php [hunk 4]
```diff
@@ -2965,6 +2979,11 @@
       'notifications_id'         =>  '67',
       'mode'                     =>  'mailing',
       'notificationtemplates_id' =>  25,
+   ], [
+      'id'                       => 68,
+      'notifications_id'         =>  '68',
+      'mode'                     =>  'mailing',
+      'notificationtemplates_id' =>  26,
    ],
```
Comment (cedric-anne) [inline install/empty_data.php]: You should also add an entry for notification `69` ('Alert domains close expiries').

File: inc/define.php [hunk 5]
```diff
@@ -476,6 +476,9 @@
 
 $CFG_GLPI['kanban_types']        = ['Project'];
 
+$CFG_GLPI['domain_types']        = ['Computer', 'Monitor', 'NetworkEquipment', 'Peripheral',
```
Comment (cedric-anne) [inline inc/define.php]: You should add a clean logic for items in `$CFG_GLPI['domain_types']` in `CommonDBTM::cleanRelationTable()`.

**Comment by cedric-anne** [comment]
To be able to associate a Ticket to a Domain or to a DomainRecord, I had to add these 2 itemtypes to `$CFG_GLPI["ticket_types"]`.

File: inc/domain.class.php [hunk 6]
```diff
@@ -41,9 +41,757 @@ class Domain extends CommonDropdown {
...
+   }
+
+   function defineTabs($options = []) {
+      $ong = [];
+      $this->addDefaultFormTab($ong);
+      $this->addStandardTab('DomainRecord', $ong, $options);
+      $this->addStandardTab('Domain_Item', $ong, $options);
+      $this->addStandardTab('Infocom', $ong, $options);
+      $this->addStandardTab('Ticket', $ong, $options);
+      $this->addStandardTab('Item_Problem', $ong, $options);
```
Comment (cedric-anne) [inline inc/domain.class.php]: I would also add Change tab.
```suggestion
      $this->addStandardTab('Item_Problem', $ong, $options);
      $this->addStandardTab('Change_Item', $ong, $options);
```

File: inc/define.php (hunk 5 above)
Comment (trasher) [reply inc/define.php]: See last commit

File: inc/domainrecord.class.php [hunk 7]
```diff
@@ -339,6 +260,29 @@ function showForm($ID, $options = []) {
       echo "</td>";
       echo "</tr>";
 
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>" . __('Data') . "</td>";
+      echo "<td colspan='3'>";
+      Html::autocompletionTextField($this, "data");
```
Comment (cedric-anne) [inline inc/domainrecord.class.php]: Some records can have lots of chars in their data (`TXT` records for example). Maybe having a textarea here could be more readable (but we will loose the auutocomplete feature).
By the way, autocomplete needs to be declared in the corresponding SO.

File: inc/domainrecord.class.php [hunk 8]
```diff
@@ -0,0 +1,483 @@
...
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class DomainRecord extends CommonDBChild {
+   const STATUS_ACTIVE = 1;
+   const STATUS_DISABLED = 0;
+   const DEFAULT_TTL = 3600;
+
+   static $rightname              = 'domain';
```
Comment (cedric-anne) [inline inc/domainrecord.class.php]: I had to add this to be able to edit a DomainRecord that is not attached to a Domain.
```suggestion
   static $rightname              = 'domain';
   static public $mustBeAttached  = false;
```

File: inc/domainrecord.class.php (hunk 7 above)
Comment (trasher) [reply inc/domainrecord.class.php]: OK, I've enabled autocompletion for now, we'll see what is the usage , this is quite easy to change

**Pull request merged**